backend/program/__init__.py

```diff
@@ -24,47 +46,209 @@ def __init__(self, args):
             self.pickly = Pickly(self.media_items, data_dir_path)
             self.pickly.start()
-        self.core_manager = ServiceManager(
-            self.media_items, True, Content, Plex, Scraping, Debrid, Symlinker
-        )
+        try:
+            self.initialize_services()
+        except Exception as e:
+            logger.error(traceback.format_exc())
+
+        # seed initial MIC with Library State
+        for item in self.services[PlexLibrary].run():
+            self.media_items.upsert(item)
+
         if self.validate():
             logger.info("Iceberg started!")
         else:
             logger.info("----------------------------------------------")
             logger.info("Iceberg is waiting for configuration to start!")
             logger.info("----------------------------------------------")
+        self.scheduler = BackgroundScheduler()
+        self.executor = ThreadPoolExecutor(thread_name_prefix="Worker") 
+        self._schedule_services()
         super().start()
+        self.scheduler.start()
         self.running = True
         self.initialized = True
 
+    def _schedule_services(self) -> None:
+        """Schedule each service based on its update interval."""
+        scheduled_services = { **self.content_services, **self.library_services }
+        for service_cls, service_instance in scheduled_services.items():
+            if not service_instance.initialized:
+                logger.info("Not scheduling %s due to not being initialized", service_cls.__name__)
+                continue
+            update_interval = service_instance.settings.update_interval
+            self.scheduler.add_job(
+                self._submit_job,
+                'interval',
+                seconds=update_interval,
+                args=[service_cls, None],
+                id=f'{service_cls.__name__}_update',
+                max_instances=1,
+                replace_existing=True,  # Replace existing jobs with the same ID
+                next_run_time=datetime.now() if service_cls != PlexLibrary else None
+            )
+            logger.info("Scheduled %s to run every %s seconds.", service_cls.__name__, update_interval)
+        return None
+
+    def _process_future_item(self, future: Future, service: Service, input_item: MediaItem) -> None:
+        """Callback to add the results from a future emitted by a service to the event queue."""
+        try:
+            for item in future.result():
+                if not isinstance(item, MediaItem):
+                    logger.error("Service %s emitted item %s of type %s, skipping", service.__name__, item, item.__class__.__name__)
+                    continue
+                self.job_queue.put(Event(emitted_by=service, item=item))
+        except Exception as e:
+            logger.error("Service %s failed with exception %s", service.__name__, traceback.format_exc())
+
+    def _service_run_item(self, func: Callable, item: MediaItem | None) -> MediaItemGenerator:

```

Comment:
This seems redundant;

future = self.executor.submit(func, item) if item is not None else self.executor.submit(func)

Reply:
Good call.  I used to do other things in here but now its just vestigial.